segment.go

```diff
@@ -53,8 +53,11 @@ func (*ZapPlugin) Open(path string) (segment.Segment, error) {
 
 	rv := &Segment{
 		SegmentBase: SegmentBase{
-			fieldsMap:      make(map[string]uint16),
-			fieldFSTs:      make(map[uint16]*vellum.FST),
+			fieldsMap: make(map[string]uint16),
+			fieldFSTs: make(map[uint16]*vellum.FST),
+			vectorCache: &vecCache{
+				cache: make(map[uint16]*cacheEntry),
```

Comment:
Rather than bleeding the cacheEntry details here, add an API - `newVectorCache(..)` that does this for you where the structs are declared.